.htaccess

```diff
@@ -1,240 +1,561 @@
+	BrowserMatchNoCase "PageGrabber(.*)" BAD_BOT
+	BrowserMatchNoCase "Papa Foto(.*)" BAD_BOT
+	BrowserMatchNoCase "pavuk(.*)" BAD_BOT
+	BrowserMatchNoCase "pcBrowser(.*)" BAD_BOT
+	BrowserMatchNoCase "python(.*)" BAD_BOT
+	BrowserMatchNoCase "Qwantify(.*)" BAD_BOT
+	BrowserMatchNoCase "RealDownload(.*)" BAD_BOT
+	BrowserMatchNoCase "ReGet(.*)" BAD_BOT
+	BrowserMatchNoCase "serpstatbot(.*)" BAD_BOT
+	BrowserMatchNoCase "SeekportBot(.*)" BAD_BOT
+	BrowserMatchNoCase "SemrushBot(.*)" BAD_BOT
+	BrowserMatchNoCase "SeznamBot(.*)" BAD_BOT
+	BrowserMatchNoCase "SiteAuditBot(.*)" BAD_BOT
+	BrowserMatchNoCase "SiteSnagger(.*)" BAD_BOT
+	BrowserMatchNoCase "SmartDownload(.*)" BAD_BOT
+	BrowserMatchNoCase "SuperBot(.*)" BAD_BOT
+	BrowserMatchNoCase "SuperHTTP(.*)" BAD_BOT
+	BrowserMatchNoCase "Surfbot(.*)" BAD_BOT
+	BrowserMatchNoCase "tAkeOut" BAD_BOT
+	BrowserMatchNoCase "Teleport Pro(.*)" BAD_BOT
+	BrowserMatchNoCase "Turnitin(.*)" BAD_BOT
+	BrowserMatchNoCase "VoidEYE(.*)" BAD_BOT
+	BrowserMatchNoCase "Web Image Collector(.*)" BAD_BOT
+	BrowserMatchNoCase "Web Sucker(.*)" BAD_BOT
+	BrowserMatchNoCase "WebAuto(.*)" BAD_BOT
+	BrowserMatchNoCase "WebCopier(.*)" BAD_BOT
+	BrowserMatchNoCase "WebFetch(.*)" BAD_BOT
+	BrowserMatchNoCase "WebGo IS(.*)" BAD_BOT
+	BrowserMatchNoCase "WebLeacher(.*)" BAD_BOT
+	BrowserMatchNoCase "WebReaper(.*)" BAD_BOT
+	BrowserMatchNoCase "WebSauger(.*)" BAD_BOT
+	BrowserMatchNoCase "Website eXtractor(.*)" BAD_BOT
+	BrowserMatchNoCase "Website Quester(.*)" BAD_BOT
+	BrowserMatchNoCase "WebStripper(.*)" BAD_BOT
+	BrowserMatchNoCase "WebWhacker(.*)" BAD_BOT
+	BrowserMatchNoCase "WebZIP(.*)" BAD_BOT
+	BrowserMatchNoCase "Wget(.*)" BAD_BOT
+	BrowserMatchNoCase "Widow(.*)" BAD_BOT
+	BrowserMatchNoCase "WWWOFFLE(.*)" BAD_BOT
+	BrowserMatchNoCase "Xaldon WebSpider(.*)" BAD_BOT
+	BrowserMatchNoCase "Zeus(.*)" BAD_BOT
+	BrowserMatchNoCase "zgrab(.*)" BAD_BOT
+	BrowserMatchNoCase "ZoominfoBot(.*)" BAD_BOT
+
+###########################################
+## Deny access (Default)
+## Comment the lines if you use the alternative method
+
+	Order allow,deny
+	Deny from env=BAD_BOT
+
+###########################################
+## Redirect to external location (Alternative)
+## Apache Module mod_rewrite must be enabled
 
-        #AuthName "Cron auth"
-        #AuthUserFile ../.htpasswd
-        #AuthType basic
-        #Require valid-user
+	#RewriteCond %{env:BAD_BOT} !^$
+	#RewriteRule ^(.*)$ https://www.example.com/ [L,R=301]
+
+</IfModule>
+
+<IfModule mod_rewrite.c>
 
 ############################################
+## The ultimate hotlink protection
+## Before use set the YOUR_DOMAIN_NAME value in the last condition 
+## https://perishablepress.com/creating-the-ultimate-htaccess-anti-hotlinking-strategy/
+
+	RewriteEngine on
+	RewriteCond %{HTTP_REFERER} !^$
+	RewriteCond %{REQUEST_FILENAME} -f
+	RewriteCond %{REQUEST_FILENAME} \.(gif|jpe?g?|png)$ [NC]
+	RewriteCond %{HTTP_REFERER} !^https?://([^.]+\.)?YOUR_DOMAIN_NAME\. [NC]
```

Comment:
Is this going to break all images by default until the site admin notices (if they notice) that they need to update `YOUR_DOMAIN_NAME` in .htaccess?

Reply:
People may not want to disable hotlinking in the first place. For example affiliates who want to use product images.

Reply:
Please read the article in the link to understand the sequence of conditions. It is the best hotlink implementation I've found so far. The author had several variants in mind that he analyzed professionally and kept only this one.

Reply:
> People may not want to disable hotlinking in the first place. For example affiliates who want to use product images.

For me it was more important to offer the best solution for those who need something like that. If people don't want to protect themselves on the hotlinking then they don't need this block, they can comment on it. If you notice there are situations where I have offered two denying methods, one classic and one based on redirection. It remains at the discretion of the users they will use.

Reply:
From the link:

> Implementing the de facto htaccess hotlink protection code required a simple binary decision: “do you want hotlink-protection or not?”

We should not make this decision for users IMO. In the past I absolutely had instructions for affiliate marketers to hot link images from my site. This would break that.

Okay but looking at the changes again I see that both RewriteRules are commented out, so it shouldn't do anything. But what happens if the user puts some other RewriteRule at the bottom of this file? Are the previous RewriteConditions affecting that? (I actually don't know here).

Reply:
> If people don't want to protect themselves on the hotlinking then they don't need this block, they can comment on it.

In general, it's best to make people opt-in to changes like this. To change it by default is the very nature of a breaking change.

Reply:
I understand your opinion. the hotlink block will be commented and whoever wants this feature will have to activate it. The important thing is that it exists as a solution.